{% extends "layout.html" %}
{% block body %}

<div class="container">
	<h1>About TGAC</h1>

<p>The Genomic Ascertainment Cohort is a data resource for the study of genetic disease while minimizing the bias of phenotypic ascertainment. An intramural investigator or authorized collaborator can query the TGAC database for a gene or variant, identify that there are one or more individuals within this dataset who harbor the variant, and then propose research projects involving that individual or those individuals. The proposed projects can have a clinical or basic science focus. For example, you might find a heterozygous nonsense variant in the dataset and want to know if that individual has associated nonsense-mediated mRNA decay of the mutant transcript. Similarly, if you have a mouse model in which the heterozygote knockout has a skin condition, you may want to interview or examine patients with a rare loss of function variant in that gene.</p>
 
<p>If you would like to propose a research project, please complete the <a href="/application">research proposal application</a>. The TGAC team will evaluate your proposal and contact you for further discussion. Once a project is accepted, the TGAC team will help re-contact the individuals identified as having variants of interest.  Proposals will be prioritized based on scientific interest of the research question as well as feasibility of the request.</p>
 
<p>Please note that not all research requests will be possible. It is our hope that all individuals in TGAC identified as having variants of interest will want to participate in post hoc projects. However, some participants may decline requests for evaluation. At this time, these data are limited to intramural research program investigators and authorized collaborators.</p>


<!--
  <p>The Genomic Ascertainment Cohort is a data resource for the study of genetic disease while minimizing the bias of phenotypic ascertainment.  An intramural investigator or authorized collaborator can query the TGAC database for a gene or variant, identify that there are one or more individuals within this dataset who harbor the variant, and then propose research projects involving that individual or those individuals.  The proposed projects can have a clinical or basic science focus.  For example, you might find a heterozygous nonsense variant in the dataset and want to know if that individual has associated nonsense-mediated mRNA decay of the mutant transcript.  Similarly, if you have a mouse model in which the heterozygote knockout has a skin condition, you may want to interview or examine patients with a rare loss of function variant in that gene.</p>
  <p>Our team will facilitate the use of TGAC for your proposed project, including the process of contacting the individuals identified as having variants of interest.  If you have a research project that you would like to propose, please complete the <a href="/application">research proposal application</a>.</p>
  <p>Please note that not all research requests will be possible.  It is our hope that all individuals in TGAC identified as having variants of interest will want to participate in post hoc projects.  However, some participants may decline requests for evaluation.  At this time, these data are limited to intramural research program investigators and authorized collaborators.</p>
-->


	<!-- <p>The concept behind <em>The Genomic Ascertainment Cohort</em> is that an intramural investigator or authorized collaborator can query the database for a gene or variant, identify that there are one or more individuals within this dataset who harbor the variant, and then propose to perform research projects on source materials from that participant (bench or basic science experiments) or on the participant themselves (clinical research experiments). For example, you might find a heterozygous nonsense variant in the dataset and want to know if that individual has associated nonsense-mediated mRNA decay of the mutant transcript. If that gene is expressed in lymphoblasts, we would provide you with an LB cell line from that participant upon which you can test your hypothesis. If you have a mouse model where the heterozygote knockout has a skin condition, we could interview or examine patients with a rare loss of function variant in that gene.</p>
	<p>If you find a variant in this dataset and have a research project that you would like to propose, please send a brief &#40;&#60;1 page&#41; proposal of the design of the experiment to Drs. Biesecker and Siegel. Note that various types of samples may already be procured on ClinSeq participants and readily available (e.g., lymphoblast cell lines), whereas others may or may not be possible to procure (e.g., fibroblasts). The proposal of clinical studies may require resources or support from the requesting investigator. We are developing and exploring how this will work and request that investigators work with us to see how these kinds of studies can practically and effectively be accomplished. Note that not all post-hoc research requests may be possible. Some participants in this dataset may be deceased, others may choose not to comply with a request for sampling or evaluation, but many will.</p>
	<p>Investigators who request biomaterials via this project may be required to sign a material transfer agreement.</p>
	<p>At this time, these data are limited to intramural research program investigators. Others should access the data through dbGAP.</p> -->
	
	<h3>Principal Investigators</h3>
	<ul>
		<li>Leslie G. Biesecker</li>
		<li>Richard Siegel</li>
	</ul>
  <h3>Clinical Team</h3>
	<ul>
		<li>Alexander Katz</li>
		<li>Katie Lewis</li>
		<li>David Ng</li>
	</ul>
	<h3>Informatics Team</h3>
	<ul>
		<li>Justin Paschall</li> 
		<li>Tyra Wolfsberg</li>
	</ul>
	<h3>Contributing Cohorts</h3>
	<ul>
		<li><a href="https://www.genome.gov/20519355/clinseq-a-largescale-medical-sequencing-clinical-research-pilot-study/" target="_blank">ClinSeq <i class="fa fa-external-link" aria-hidden="true"></i></a></li>
	</ul>
        <br>
        <p><h4>Questions</strong></h4>
        <p>Please contact <a href="mailto:tgac@mail.nih.gov">tgac@mail.nih.gov</a></p>
 

	<!-- <h3>Contributors to the TGAC Web Site</h3>
	<ul>
		<li>Tyra Wolfsberg</li>
		<li>Anh-Dao Nguyen</li>
		<li>Travis Moreland</li>
		<li>Jamie Castle</li>
		<li>Mark Fredriksen</li>
		<li>Justin Paschall</li>
	</ul> -->
</div>


<!--
    <div class="container">
        <h1>About gnomAD</h1>
        <div style="font-size: 16px;">
            <div>
                <p>
                    The Genome Aggregation Database (gnomAD), is a coalition of investigators seeking to aggregate and harmonize exome and genome sequencing data from a variety of large-scale sequencing projects, and to make summary data available for the wider scientific community. In its first release, which contained exclusively exome data,  it was known as the Exome Aggregation Consortium (ExAC).
                </p>
                <p>
                    The data set provided on this website spans 123,136 exomes and 15,496 genomes from unrelated individuals sequenced as part of various disease-specific and population genetic studies. <a href="https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad">This blog post</a> describes the latest release. We have removed individuals known to be affected by severe pediatric disease, as well as their first-degree relatives, so this data set should serve as a useful reference set of allele frequencies for severe disease studies - however, note that some individuals with severe disease may still be included in the data set, albeit likely at a frequency equivalent to or lower than that seen in the general population.

                </p>
                <p>
                    All of the raw data from these projects have been reprocessed through the same pipeline, and jointly variant-called to increase consistency across projects. The processing pipelines were written in the <a href="https://software.broadinstitute.org/wdl/">WDL workflow definition language</a> and executed using the <a href="https://github.com/broadinstitute/cromwell">Cromwell execution engine</a>, open-source projects for defining and executing genomic workflows at massive scale on multiple platforms. The gnomAD data set contains individuals sequenced using multiple exome capture methods and sequencing chemistries, so coverage varies between individuals and across sites. This variation in coverage is incorporated into the variant frequency calculations for each variant.
                </p>
                <p>
                    gnomAD was QCed and analysed using the <a href="https://hail.is/">Hail</a> open-source framework for scalable genetic analysis.
                </p>
                <p>
                    A list of gnomAD Principal Investigators and groups that have contributed data and analysis to the current release is available below.
                </p>
                <p>
                    The generation of this call set was funded primarily by the Broad Institute, and the data here are released publicly for the benefit of the wider biomedical community. There are no publication restrictions or embargoes on these data. Please cite the <a href="http://www.nature.com/nature/journal/v536/n7616/full/nature19057.html">ExAC paper</a> for any use of these data.
                </p>
                <p>
                    The data are available under the <a href="http://opendatacommons.org/licenses/odbl/1.0/">ODC Open Database License (ODbL)</a> (summary available <a href="http://www.opendatacommons.org/licenses/odbl/1-0/summary/">here</a>): you are free to share and modify the gnomAD data so long as you attribute any public use of the database, or works produced from the database; keep the resulting data-sets open; and offer your shared or adapted version of the dataset under the same ODbL license.
                </p>
                <p> The aggregation and release of summary data from the exomes and genomes collected by the Genome Aggregation Database has been approved by the Partners IRB
		  (protocol 2013P001339, "Large-scale aggregation of human genomic data").
                </p>
            </div>
            <p>
                For bug reports, please file an issue on <a href="https://github.com/macarthur-lab/gnomad_browser/issues">Github</a>.
            </p>
        </div>
        <div class="credits">
            <div class="col-md-2 credit-section">
                <h4>Principal Investigators</h4>
                <ul class="credit-pi-first">
                    <li>Daniel MacArthur</li>
                    <li>Aarno Palotie</li>
                    <li>Andres Metspalu</li>
                    <li>Anne Remes</li>
                    <li>Adolfo Correa</li>
                    <li>Andre Franke</li>
                    <li>Ann Pulver</li>
                    <li>Ben Glaser</li>
                    <li>Ben Neale</li>
                    <li>Bong-Jo Kim</li>
                    <li>Carlos Pato</li>
                    <li>Carlos A Aguilar Salinas</li>
                    <li>Christina Hultman</li>
                    <li>Christine M. Albert</li>
                    <li>Christopher Haiman</li>
                    <li>Clicerio Gonzalez</li>
                    <li>Colin Palmer</li>
                    <li>Craig Hanis</li>
                    <li>Dan Roden</li>
                    <li>Dan Turner</li>
                    <li>Dana Dabelea</li>
                    <li>Daniel Chasman</li>
                    <li>Danish Saleheen</li>
                    <li>David Altshuler</li>
                    <li>David Goldstein</li>
                    <li>Dawood Darbar</li>
                    <li>Dermot McGovern</li>
                    <li>Diego Ardissino</li>
                    <li>Donald Bowden</li>
                    <li>Emelia J. Benjamin</li>
                    <li>Erkki Vartiainen</li>
                    <li>Erwin Bottinger</li>
                    <li>Gad Getz</li>
                    <li>George Kirov</li>
                    <li>Gil Atzmon</li>
                    <li>Harlan M. Krumholz</li>
                    <li>Harry Sokol</li>
                    <li>Heribert Schunkert</li>
                    <li>Hilkka Soininen</li>
                    <li>Hugh Watkins</li>
                    <li>Jaakko Kaprio</li>
                    <li>Jaana Suvisaari</li>
                    <li>James Meigs</li>
		    <li>James Ware</li>
                    <li>James Wilson</li>
                    <li>Jaspal Kooner</li>
                    <li>Jaume Marrugat</li>
                    <li>Jeanette Erdmann</li>
                    <li>Jeremiah Scharf</li>
                    <li>John Barnard</li>
                    <li>John Chambers</li>
                    <li>John D. Rioux</li>
                    <li>Jose Florez</li>
                    <li>Josée Dupuis</li>
                    <li>Judy Cho</li>
                </ul>
            </div>
            <div class="col-md-2 credit-section">
                <ul class="credit-pi-second">
                    <li>Juliana Chan</li>
                    <li>Kyong Soo Park</li>
                    <li>Leif Groop</li>
                    <li>Lorena Orozco</li>
                    <li>Lori Bonnycastle</li>
                    <li>Maija Wessman</li>
                    <li>Mark Daly</li>
                    <li>Mark McCarthy</li>
                    <li>Markku Laakso</li>
                    <li>Martti Färkkilä</li>
                    <li>Matthew Bown</li>
                    <li>Matthew Harms</li>
                    <li>Matti Holi</li>
                    <li>Michael Boehnke</li>
                    <li>Michael O'Donovan</li>
                    <li>Michael Owen</li>
                    <li>Mikko Hiltunen</li>
                    <li>Mikko Kallela</li>
                    <li>Mina Chung</li>
                    <li>Ming Tsuang</li>
                    <li>Moore Shoemaker</li>
                    <li>Nazneen Rahman</li>
                    <li>Nilesh Samani</li>
                    <li>Olle Melander</li>
                    <li>Pamela Sklar</li>
                    <li>Patrick T. Ellinor</li>
                    <li>Patrick Sullivan</li>
		    <li>Peter Nilsson</li>
                    <li>Ramnik Xavier</li>
                    <li>Ravindranath Duggirala</li>
                    <li>Rinse Weersma</li>
                    <li>Roberto Elosua</li>
                    <li>Ronald Ma</li>
                    <li>Ruth Loos</li>
                    <li>Ruth McPherson</li>
                    <li>Samuli Ripatti</li>
                    <li>Sekar Kathiresan</li>
                    <li>Seppo Koskinen</li>
                    <li>Soo Heon Kwak</li>
                    <li>Stephen Glatt</li>
                    <li>Steve McCarroll</li>
                    <li>Steven A. Lubitz</li>
                    <li>Subra Kugathasan</li>
                    <li>Tai Shyong</li>
                    <li>Tariq Ahmad</li>
                    <li>Teresa Tusie Luna</li>
                    <li>Terho Lehtimäki</li>
                    <li>Tim Spector</li>
                    <li>Tõnu Esko</li>
                    <li>Tuomi Tiinamaija</li>
                    <li>Veikko Salomaa</li>
                    <li>Yik Ying Teo</li>
                    <li>Young Jin Kim</li>
                </ul>
            </div>
            <div class="col-md-4 credit-section clearfix">
                <h4>Contributing projects</h4>
                <ul>
                    <li>1000 Genomes</li>
                    <li>1958 Birth Cohort</li>
                    <li>ALSGEN</li>
                    <li>Alzheimer's Disease Sequencing Project (ADSP)</li>
                    <li>Atrial Fibrillation Genetics Consortium (AFGen)</li>
                    <li>Estonian Genome Center, University of Tartu (EGCUT)</li>
                    <li>Bulgarian Trios</li>
                    <li>Finland-United States Investigation of NIDDM Genetics (FUSION)</li>
                    <li>Finnish Twin Cohort Study</li>
                    <li>FINN-ADGEN</li>
                    <li>FINRISK</li>
                    <li>Framingham Heart Study</li>
                    <li>Génome Québec - Genizon Biobank</li>
                    <li>Genomic Psychiatry Cohort</li>
                    <li>GoT2D</li>
                    <li>Genotype-Tissue Expression Project (GTEx)</li>
                    <li>Health2000</li>
                    <li>Inflammatory Bowel Disease:</li>
                    <ul class="credit-subset">
                        <li>Helsinki University Hospital Finland</li>
                        <li>NIDDK IBD Genetics Consortium</li>
                        <li>Quebec IBD Genetics Consortium</li>
                    </ul>
                    <li>Jackson Heart Study</li>
                    <li>Kuopio Alzheimer Study</li>
                    <li>LifeLines Cohort</li>
                    <li>MESTA</li>
                    <li>METabolic Syndrome In Men (METSIM)</li>
                    <li>Finnish Migraine Study</li>
                    <li>Myocardial Infarction Genetics Consortium (MIGen):</li>
                    <ul class="credit-subset">
                        <li>Leicester Exome Seq</li>
		        <li>North German MI Study</li>
                        <li>Ottawa Genomics Heart Study</li>
                        <li>Pakistan Risk of Myocardial Infarction Study (PROMIS)</li>
                        <li>Precocious Coronary Artery Disease Study (PROCARDIS)</li>
                        <li>Registre Gironi del COR (REGICOR)</li>
		        <li>South German MI Study</li>
			<li>Variation in Recovery: Role of Gender on Outcomes of Young AMI Patients (VIRGO)</li>
                    </ul>
                    <li>National Institute of Mental Health (NIMH) Controls</li>
                    <li>NHLBI-GO Exome Sequencing Project (ESP)</li>
		    <li>NHLBI TOPMed</li>
                    <li>Schizophrenia Trios from Taiwan</li>
                    <li>Sequencing Initiative Suomi (SiSu)</li>
                    <li>SIGMA-T2D</li>
                    <li>Swedish Schizophrenia & Bipolar Studies</li>
                    <li>T2D-GENES</li>
		    <ul class="credit-subset">
			    <li>GoDARTS</li>
		    </ul>
                    <li>T2D-SEARCH</li>
			<li>The Cancer Genome Atlas (TCGA)</li>
                </ul>

            </div>
            <div class="col-md-2 credit-section">
                <h4>Production team</h4>
                <ul>
                    <li>Eric Banks</li>
                    <li>Charlotte Tolonen</li>
		    <li>Christopher Llanwarne</li>
                    <li>Dave Shiga</li>
                    <li>Fengmei Zhao</li>
		    <li>Jeff Gentry</li>
                    <li>Jose Soto</li>
                    <li>Kathleen Tibbetts</li>
		    <li>Khalid Shakir</li>
		    <li>Kristian Cibulskis</li>
                    <li>Laura Gauthier</li>
		    <li>Miguel Covarrubias</li>
                    <li>Monkol Lek</li>
                    <li>Ryan Poplin</li>
		    <li>Ruchi Munshi</li>
                    <li>Sam Novod</li>
		    <li>Thibault Jeandet</li>
                    <li>Valentin Ruano-Rubio</li>
                    <li>Yossi Farjoun</li>
                </ul>
                <h4>Analysis team</h4>
                <ul>
                    <li>Konrad Karczewski</li>
                    <li>Laurent Francioli</li>
                    <li>Kristen Laricchia</li>
                    <li>Monkol Lek</li>
                    <li>Anne O'Donnell Luria</li>
		    <li>Ben Neale</li>
                    <li>Beryl Cummings</li>
		    <li>Cotton Seed</li>
                    <li>Daniel Birnbaum</li>
                    <li>Eric Minikel</li>
                    <li>James Ware</li>
                    <li>Kaitlin Samocha</li>
		    <li>Laramie Duncan</li>
		    <li>Mark Daly</li>
                    <li>Tim Poterba</li>
                </ul>
            </div>
            <div class="col-md-2 credit-section">
                <h4>Website team</h4>
                <ul>
                    <li>Ben Weisburd</li>
                    <li>Konrad Karczewski</li>
                    <li>Matthew Solomonson</li>
                    <li>Daniel Birnbaum</li>
                </ul>
                <h4>Ethics team</h4>
                <ul>
                    <li>Jessica Alföldi</li>
                    <li>Andrea Saltzman</li>
                    <li>Molly Schleicher</li>
                    <li>Namrata Gupta</li>
                    <li>Stacey Donnelly</li>
                </ul>
                <h4>Broad Genomics Platform</h4>
                <ul>
                    <li>Stacey Gabriel</li>
		    <li>Kristen Connolly</li>
                    <li>Steven Ferriera</li>

                </ul>
                <h4>Funding</h4>
                <p>NIGMS R01 GM104371<br> (PI: MacArthur)</p>
                <p>NIDDK U54 DK105566<br> (PIs: MacArthur and Neale)</p>
                <p>The vast majority of the data storage, computing resources, and human effort used to generate this call set were donated by the Broad Institute.</p>
            </div>
        </div>
    </div>

    -->

{% endblock %}
