{% extends "layout.html" %}
{% block body %}

<script>
	$('#navbar_form_container').hide();
</script>

<div class="container">
	<h1 id="home-title">The Genome Ascertainment Cohort (TGAC) Browser <sup style="color: red; font-weight: 300; font-style: italic;">Beta</sup></h1>
	<div class="row">
		<div class="col-lg-12">
			<div id="home-searchbox">
				<form action="/awesome" role="search">
					<input type="submit" style="display: none;"/>
					<input type="text" id="home-searchbox-input" name="query" class="form-control input-lg awesomebar" maxlength="100" placeholder="Search for a gene, transcript, variant, or region"/>
					<input type="submit" style="position: absolute; left: -9999px"/>
				</form>
				<p class="text-muted">Examples - Gene: <a href="/awesome?query=PCSK9">PCSK9</a>, Variant: <a href="/variant/1-55505647-G-T">1-55505647-G-T</a></p>
			</div>
		</div>
	</div>
        <!--
        <div class="row" style="font-size: 16px;">
            <div class="col-md-8">
                <h3 align="center"> <a href="/about" style="color: black;">About gnomAD</a></h3>
                <div>
                    <p>
                        The <a href="/about">Genome Aggregation Database</a> (gnomAD) is a resource developed by an international coalition of investigators, with the goal of aggregating and harmonizing both exome and genome sequencing data from a wide variety of large-scale sequencing projects, and making summary data available for the wider scientific community.
                    </p>
                    <p>
                        The data set provided on this website spans 123,136 exome sequences and 15,496 whole-genome sequences from unrelated individuals sequenced as part of various disease-specific and population genetic studies. The gnomAD Principal Investigators and groups that have contributed data to the current release are listed <a href="/about">here</a>.
                    </p>
                    <p>
                        All data here are released for the benefit of the wider biomedical community, without restriction on use - see the terms of use <a href="/terms">here</a>.
                    </p>
                    <p>
                        Sign up for our mailing list for future release announcements <a href="https://groups.google.com/forum/#!forum/exac_data_announcements">here</a>.
                    </p>
                </div>
            </div>
            <div class="col-md-4">
                <h3 align="center">Recent News</h3>
                <h4>October 3, 2017</h4> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/">gnomAD r2.0.2</a> released. Sample composition is identical to the previous release (r2.0.1), however we have
                made a change to the variant filtering process that you can read about <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/README.txt">here</a>.
                <h4>February 27, 2017</h4> <a href="https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad">Official gnomAD release (version 2.0)</a> with browser updates and data available for <a href="/downloads">download</a>.
                <h4>October 19, 2016</h4> Public release of gnomAD Browser (beta) at ASHG!

            </div>
        </div>
    </div>
     -->

	<div class="row">
		<div class="col-md-8">
			<!-- <h3><a href="/about" style="color: black;"><em><strong>The Genomic Ascertainment Cohort</strong></em> (TGAC) Browser</a></h3> -->
			<div>
                         <p>The Genome Ascertainment Cohort (TGAC) is a data resource that aims to improve our understanding of the phenotypic consequence of genetic variation. The project is designed to enable further study of individuals via genomic ascertainment without prior knowledge of phenotype. TGAC is a collaboration between NHGRI (<a href="https://irp.nih.gov/pi/leslie-biesecker" target="_blank">Les Biesecker <i class="fa fa-external-link" aria-hidden="true"></i></a> ) and NIAMS (<a href="https://irp.nih.gov/pi/richard-siegel" target="_blank">Richard Siegel <i class="fa fa-external-link" aria-hidden="true"></i></a> ).</p>
 
<p>Investigators may use this web interface to view aggregate exome and genome data from patients in contributing cohorts.  Patients in TGAC have agreed to be re-contacted for secondary phenotyping studies.  After identifying a variant or variants of interest, investigators may propose projects to study phenotypes that may be associated with the genotype.  The initial dataset included in this beta release of the TGAC Browser comprises exomes from 938 participants in the NHGRI-sponsored ClinSeq® study.  We plan to add more exome and genome data from other investigators soon.  This beta release does not contain variants in genes on the X or Y chromosome, but these data will be added soon.  See <a href="/about" target="_blank">&ldquo;About <i>The Genomic Ascertainment Cohort</i>&rdquo;</a> for more information.</p>
 
<p>All data here are released under a <a href="https://www.genome.gov/pages/research/wellcomereport0303.pdf"  target="_blank">Fort Lauderdale Agreement</a> for the benefit of the IRP community. Please read and understand that agreement before browsing the variants.</p>
 
<p>We thank Konrad Karczewski, Daniel MacArthur, and the gnomAD browser team for the original <a href="https://github.com/macarthur-lab/gnomad_browser" target="_blank">gnomAD browser code  <i class="fa fa-external-link" aria-hidden="true"></i></a>  modified to create this resource.</p>


                       
<!--
				<p>TGAC is a data resource that aims to improve our understanding of the phenotypic consequence of genetic variation.  Exome and genome data from patients in contributing cohorts are available to view in aggregate via this browser.  This project is designed to enable further study of individuals via genomic ascertainment without prior knowledge of phenotype.</p>
				<p>The initial dataset included in this beta release of the TGAC Browser comprises exomes from 938 participants in the NHGRI-sponsored ClinSeq&reg; study.  We plan to add more exome and genome data from other investigators soon.  See <a href="/about">“About The Genomic Ascertainment Cohort”</a> for information about this effort.  This project is a collaboration between NHGRI (<a href="https://irp.nih.gov/pi/leslie-biesecker" target="_blank">Les Biesecker <i class="fa fa-external-link" aria-hidden="true"></i></a>) and NIAMS (<a href="https://irp.nih.gov/pi/richard-siegel" target="_blank">Richard Siegel <i class="fa fa-external-link" aria-hidden="true"></i></a>).</p>
				<p>All data here are released under a <a href="https://www.genome.gov/pages/research/wellcomereport0303.pdf">Fort Lauderdale Agreement <i class="fa fa-external-link" aria-hidden="true"></i></a> for the benefit of the IRP community. Please read and understand that agreement before using the data.</p>
				<p>Our team will facilitate the use of TGAC for your proposed project, including the process of contacting the individuals identified as having variants of interest.  If you have a research project that you would like to propose, please complete the <a href="/application">research proposal application</a>.</p> 
				<p>We thank Konrad Karczewski, Daniel MacArthur, and the gnomAD browser team for the original <a href="https://github.com/macarthur-lab/gnomad_browser" target="_blank">gnomAD browser code  <i class="fa fa-external-link" aria-hidden="true"></i></a> modified to create this resource.</p>
-->


				<!--
				<p>This  project is designed to make exome and genome data from intramural patient cohorts as well as external collaborators such as INOVA more widely available and to facilitate genomic ascertainment research studies, both basic and clinical, within the Intramural Research Program (IRP).</p>
				<p>The initial dataset included in this beta release of the TGAC Browser comprises exomes from 938 participants in the NHGRI-sponsored ClinSeq<sup>&reg;</sup> study<sup>&#9733;</sup>. Nearly all of them are unrelated. We plan to add more exome and genome data from other investigators soon. Our goal here is to pilot approaches to genomic ascertainment research studies in the IRP. See <a href="/about" target="_blank">&ldquo;About <i>The Genomic Ascertainment Cohort</i>&rdquo;</a> for information about this effort. This project is a collaboration between NHGRI (Les Biesecker) and  NIAMS (Richard Siegel).</p>
				<p>All data here are released under a <a href="https://www.genome.gov/pages/research/wellcomereport0303.pdf">Fort Lauderdale Agreement</a> for the benefit of the IRP community. Please read and understand that agreement before using the data.</p>
				<p>NIH gratefully acknowledges Daniel MacArthur and the Broad Institute for sharing the gnomAD  web architecture with us to support this pilot.</p>
				<p class="note"><sup>&#9733;</sup> Note that ClinSeq data are also available to researchers through dbGAP. The initial publication describing ClinSeq is PMID 19602640 &#38; PMCID PMC2752125</p>
					-->

			</div>
		</div>

		<div class="col-md-4">
			<h3 align="center">Recent News</h3>
			<p>Initial release of the TGAC beta browser with the ClinSeq dataset (938 exomes) - <span class="date-news">Feb 28, 2018</span></p>
                        <p>TGAC is announced and described in the March-April  issue of the NIH Catalyst. - <span class="date-news">Mar 3, 2018</span></p>
                        <p><a href="https://www.genomeweb.com/informatics/ambitious-tgac-collaboration-looks-re-engineer-genomics-research-predict-phenotypes#.Ws9kLtMbPVt"> Ambitious TGAC Collaboration Looks to Re-Engineer Genomics Research to Predict Phenotypes</a> - <span class="date-news">Mar 28, 2018</span></p> 
		</div>
	</div>
</div>

{% endblock %}
