<script type="text/javascript">
    window.is_gene = typeof window.transcripts_in_gene != 'undefined';
    window.filter_definitions = {
        SEGDUP: 'in a segmental duplication region',
        LCR: 'in a low complexity region',
        InbreedingCoeff: 'has an inbreeding coefficient < -0.3',
        RF: 'failed random forests filters',
        AC0: 'allele count is zero (i.e. no high-confidence genotype)'
    }
</script>
<script type="text/template" id="variant-table-template">
<span>
    <a class="btn btn-info hidden-xs" id="export_to_csv">Export table to CSV</a>
</span>
{% if transcripts_in_gene is defined %}
    <div>
        <small>&dagger; denotes a consequence that is for a non-canonical transcript</small>
    </div>
{% endif %}
<table id="variant_table" class="tablesorter">
    <thead>
        <tr>
            <th id="th-variantid" class="tooltip-table-header omit_csv" data-tooltip="chr:pos reference/alternate &#xa;(dbSNP135 rsID)">
                Variant</th>
            <th class='hidden'>Chrom</th>
            <th class='hidden'>Position</th>
            <td class="hidden">RSID</td>
            <th class="hidden">Reference</th>
            <th class="hidden">Alternate</th>
            <th id="th-data" class="tooltip-table-header" data-tooltip="Dataset/VQSR sensitivity filter & hard filters">Source</th>
            <th class="hidden" id="th-filterexomes-csv">Filters - exomes</th>
            <th class="hidden" id="th-filtergenomes-csv">Filters - genomes</th>
            <th id="th-hgvs" class="tooltip-table-header" data-tooltip="HGVS annotation (protein change, where defined, or transcript change for splice variants, otherwise empty)">Consequence</th>
            <th class="hidden">Protein Consequence</th>
            <th class="hidden">Transcript Consequence</th>
            <th id="th-vep" class="tooltip-table-header" data-tooltip="Variant Effect Predictor (VEP) &#xa; annotation using Gencode 81.&#xa; Worst across all transcripts of this gene.">Annotation</th>
            <th id="th-flags" class="tooltip-table-header" data-tooltip="Various flags (LoF from LOFTEE, MNP) &#xa; that may change annotation/confidence">Flags</th>
            <th id="th-ac" class="tooltip-table-header" data-tooltip="Alternate allele count in genotypes &#xa;(genotype quality >=20 & depth >=10)">Allele Count</th>
            <th id="th-an" class="hidden-xs tooltip-table-header" data-tooltip="Total number of called genotypes &#xa;(genotype quality >= 20 & depth >=10)">Allele Number</th>
            {% if chrom != 'Y' %}
                <th id="th-hom" class="hom-title homhemi hidden-xs tooltip-table-header" data-tooltip="Number of homozygous individuals for this alternate allele">Number of Homozygotes</th>
            {% endif %}
            {% if chrom == 'X' or chrom == 'Y' %}
                <th id="th-hem" class="hidden-xs tooltip-table-header" data-tooltip="Number of hemizygous individuals for this alternate allele">Number of Hemizygotes</th>
            {% endif %}
            <th id="th-af" class="tooltip-table-header allele_frequency_column" data-tooltip="Frequency using only &#xa; high-quality genotypes &#xa; The graphic displays &#xa; allele frequency on a  &#xa; discrete scale: Singletons, &#xa; <1/10,000, <1/1000, <1%, &#xa; <5%, <50%, >50%">
                Allele Frequency</th>


            <% if ( table_variants[0].pop_acs) {
                _.each(Object.keys(table_variants[0].pop_acs), function(pop) { %>
                <th class="hidden">Allele Count <%= pop %></th>
                <th class="hidden">Allele Number <%= pop %></th>
                <th class="hidden">Homozygote Count <%= pop %></th>
            <% }); } %>



        </tr>
    </thead>
    <tbody>
    <% _.each(table_variants, function(variant) { %>
        <tr class="table_variant" id="variant_<%= variant.variant_id %>"
            category="<%= variant.category %>"
            filter_status="<%= variant.filter %>"
            frequency="<%= variant.allele_freq %>"
            major_consequence="<%= variant.major_consequence %>"
            indel="<%= variant.indel %>"
            datasets="<%= variant.datasets %>"
            variant_id="<%= variant.variant_id %>"
          >
            <td id="td-variantid" class="omit_csv table-variant-id">
                <a href="/variant/<%= variant.chrom %>-<%= variant.pos %>-<%= variant.ref %>-<%= variant.alt %>" target="_blank">
                <%= variant.chrom %>:<%= variant.pos %>
                <% if (variant.ref.length > 20) { %>
                    <%= variant.ref.slice(0,20) %>...
                <% } else { %>
                    <%= variant.ref %>
                <% } %> /
                <% if (variant.alt.length > 20) { %>
                    <%= variant.alt.slice(0,20) %>...
                <% } else { %>
                    <%= variant.alt %>
                <% } %>
                <% if (variant.rsid != '.') { %>
                    (<%= variant.rsid %>)
                <% } %>
                </a>
            </td>
            <td class='hidden'> <%= variant.chrom %></td>
            <td class='hidden'> <%= variant.pos %></td>
            <td class="hidden"><%= variant.rsid %></td>
            <th class="hidden"> <%= variant.ref %></th>
            <th class="hidden"> <%= variant.alt %></th>
            <td id="td-data" class='data-indicator'>
                <div class='data-indicators'>
                    <% _.each(variant.datasets, function(dataset) { %>
                        <%
                        function spanColour(filter) {
                            if (filter === 'PASS') {
                                return 'success'
                            } else {
                                return 'warning'
                            }
                        }
                        if (dataset === 'all') return
                        if (dataset == 'ExAC') {
                            span_tag = spanColour(variant.ExAC.filter);
                            avatar =  'E';
                        } else {
                            span_tag = spanColour(variant.gnomAD.filter);
                            avatar = 'G';
                        }
                        %>
                        <div class="label data-indicator-child data-indicator-<%= dataset %> label-<%= span_tag %> tooltip-table-header" data-tooltip="This variant is found in the <%= dataset == 'ExAC' ? 'exome' : 'genome'%> dataset <%= variant[dataset].filter === 'PASS' ? 'with filter value PASS' : 'but is filtered for the following reasons: &#xa;' %> <%= variant[dataset].filter.split('|').map(function(filter_name) { if (filter_name  !== 'PASS') { return filter_name + ': ' + window.filter_definitions[filter_name] } }).join('&#xa;') %>"><%= avatar %></div>
                    <% }); %>
                </div>
            </td>
            <td class="hidden" id="td-filterexomes-csv">
                <% if (variant.ExAC) { %>
                    <%= variant.ExAC.filter %><% } else { %>
                    <%= "NA" %><% } %>
            </td>
            <td class="hidden" id="td-filtergenomes-csv">
                <% if (variant.gnomAD) { %>
                    <%= variant.gnomAD.filter %><% } else { %>
                    <%= "NA" %><% } %>
            </td>
            <td id="td-hgvs" class='table-hgvs'>
                <% if (variant.HGVS) { %>
                    <% if (variant.HGVS.length > 20) { %>
                        <%= variant.HGVS.slice(0,20) %>...<% } else { %>
                        <%= variant.HGVS %><% } %><% if (is_gene & variant.CANONICAL != 'YES') { %><span class="tooltip-table-header" data-tooltip="Annotation is for non-canonical transcript">&dagger;</span><% } %>
                <% } %>
            </td>
            <td class='hidden'> <%= variant.HGVSp %></td>
            <td class='hidden'> <%= variant.HGVSc %></td>
            <td id="td-vep" class="<%= variant.category %> table-maj-conseq" data-consequence="<%= variant.major_consequence %>"><b>
                <% if (variant.major_consequence) { %>
                    <%= variant.major_consequence.replace('_variant', '').replace(/_/g, ' ').replace('utr', 'UTR').replace('3 prime', "3'").replace('5 prime', "5'").replace('nc ', "non-coding ") %>
                <% } %>
            </b> </td>
            <td id="td-flags">
                <% _.each(variant.flags, function(flag) { %>
                    <% if (flag == 'LoF flag') { span_tag = 'warning' } else { span_tag = 'danger' } %>
                    <a href="/variant/<%= variant.chrom %>-<%= variant.pos %>-<%= variant.ref %>-<%= variant.alt %>" target="_blank"><span class="label label-<%= span_tag %> tooltip-table-header" data-tooltip="Variant annotation dubious: &#xa; see variant page for details"><%= flag %></span></a>
                <% }); %>
            </td>
            <td id="td-ac" class='table-allele-count'> <%= variant.allele_count %> </td>
            <td id="td-an" class='table-allele-num hidden-xs'> <%= variant.allele_num %> </td>
            {% if chrom != 'Y' %}
                <td id="td-hom" class="table-hom-count hidden-xs"> <%= variant.hom_count %> </td>
            {% endif %}
            {% if chrom == 'X' or chrom == 'Y' %}
                <td id="td-hem" class=" table-hemi-count hidden-xs"> <%= variant.hemi_count %> </td>
            {% endif %}
            <td id="td-af">
                <div class='frequency-container'>
                    <div class='table-allele-freq'>
                        <% if (variant.allele_freq) { %>
                            <%= variant.allele_freq.toPrecision(4) %>
                        <% } else { %>
                            <%= variant.allele_freq %>
                        <% } %>
                    </div>
                    <div class="table-allele-freq-box" id="variant_af_box_<%= variant.variant_id.replace('*', 'star') %>" style="float: right;"></div>
                </div>
            </td>

        </tr>
    <% }); %>
    </tbody>
</table>
</script>
